src/main/java/be/cytomine/client/models/Software.java

```diff
@@ -29,4 +31,21 @@ public Software(String name, String serviceName, String resultType, String execu
         this.set("resultName", resultType);
         this.set("executeCommand", executeCommand);
     }
+    public Software deprecateSoftware(Long id) throws CytomineException {
+        Software software = this.fetch(id) ;
+        software.set("deprecated", true);
+        return update();
+    }
+
+    public Software addSoftware(String softwareVersion, String name, Long idSoftwareUserRepository, Long idDefaultProcessingServer, String resultType, String executeCommand, String pullingCommand) throws CytomineException {
```

Comment:
Old way to use the Java client. Merge with the constructor which has to be used to create new objects (first locally and then saved on the server with `save()` method)

Reply:
done